### PR TITLE
Decorate a class

### PR DESCRIPTION
**Pull request by vvoo21**
### In this milestone, I performed the following project requirements ⚙️👩‍💻 :
- Create a class Nameable. with a method called correct_name that will raise a NotImplementedError.
- Make Person class inherits from Nameable and add a method correct_name which returns the name attribute.
- Create BaseDecorator class that inherits from Nameable.
- In the constructor assign a nameable object from params to an instance variable and Implement the correct_name method that returns the result of the correct_name method of the @nameable.
- Create CapitalizeDecorator class that inherits from BaseDecorator and Implement a method correct_name that capitalizes the output of @nameable.correct_name.
- Create TrimmerDecorator class that inherits from BaseDecorator and Implement a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

File: trimmer_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require './base_decorator'
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name[0...10]
+  end
+end
```
Comment (Owusu-Desmond) [inline trimmer_decorator.rb]: - [ ] Good to implement the method `correct_name` but make sure that it output of `@nameable.correct_name` has a maximum of 10 characters so I think the index range should be `[0...9]` not `[0...10]` and only If the word is longer than 10 hence you also need to implement if statement here to handle that. like this :point_down: 
  
  ```ruby
   # The name has a maximum of 10 characters. If it's longer it should trim the word.
      def correct_name
        @nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name
      end
  ```

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @vvoo21 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by vvoo21** [comment]
Thanks @Owusu-Desmond

File: trimmer_decorator.rb (hunk 1 above)
Comment (peterrobert) [reply trimmer_decorator.rb]: This change has been done well

**Review by peterrobert — APPROVED** [review]
Hi @vvoo21, 


### Highlights!

[ ] Good feature branch name ✔️
[ ] Descriptive pull request ✔️


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread or tag me @peterrobert if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**